proxy/proxy/dial.go

```diff
@@ -78,8 +84,10 @@ type Dialer func(net, addr string) (net.Conn, error)
 func Init(auth *http.Client, connset *ConnSet, dialer Dialer) {
 	dialClient.Lock()
 	dialClient.c = &Client{
-		Port:   port,
-		Certs:  certs.NewCertSource("", auth, true),
+		Port: port,
+		Certs: certs.NewCertSourceOpts(auth, certs.RemoteOpts{
+			GenerateKeyOnConnect: GenerateKeyOnConnect,
```

Comment:
This defaults to false, so is it necessary to include it here? Would it be better to have the user use `InitWithClient` to configure the client instead?

Reply:
Done, but I had to make `Port` an exported constant so users can see it. Alternately I could set `Client.Port` in `InitWithClient()` if it's zero? The cleaner solution might be an `Init` function for `Client`.